### DISCUSSION TITLE
v1.6.0

### DISCUSSION THREAD

**Discussion by romange** [dragonflydb/dragonfly#1532] [Announcements]
# Dragonfly v1.6.0

New Dragonfly release! Prominent changes include:
* Support for s390x (Thanks, @iko1 )
* Fixing stability issues in the transactional framework around blocking operations (thanks to @glitch user on Discord for helping to pinpoint the issues)
* Support plain replication on the admin port with TLS enabled on the primary port (#1490 )
* Support custom permissioning on UNIX domain sockets (#1415)
* Fix a deadlock when performing the replication over the admin port (#1470)
* Atomic replica failover support.

## What's Changed
* feat: add s390x architecture support by @iko1 in https://github.com/dragonflydb/dragonfly/pull/1214
* feat(CI): Update weekly build image name to be `dragonfly-weekly` by @Pothulapati in https://github.com/dragonflydb/dragonfly/pull/1384
* fix(docker-release): Run Docker test only if image is built by @Pothulapati in https://github.com/dragonflydb/dragonfly/pull/1430
* feat(cluster): Implement `CLUSTER KEYSLOT` command by @chakaz in https://github.com/dragonflydb/dragonfly/pull/1413
* fix(cluster): Fix race when setting cluster config by @chakaz in https://github.com/dragonflydb/dragonfly/pull/1428
* fix: out of bounds typo by @romange in https://github.com/dragonflydb/dragonfly/pull/1434
* chore: improve connection I/O errors reporting by @romange in https://github.com/dragonflydb/dragonfly/pull/1436
* test(cluster): Extend cluster_mgr.py by @chakaz in https://github.com/dragonflydb/dragonfly/pull/1426
* feat(server): set allowed key size 256MB by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1435
* feat: Implement KNN interface by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1412
* fix: add numpy to regtests (needed for knn) by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1454
* fix: call NotifyPending only from tx queue invocations by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1439
* feat(server): add flag to set FiberQueueThreadPool thread size by @wayslog in https://github.com/dragonflydb/dragonfly/pull/1452
* feat(server): add unixsocketperm support by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1450
* feat(server): Add missing fields to INFO PERSISTENCE command (#1408) by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1438
* feat(regression github): send failed tests names to chat  by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1459
* fix: rename memcache to memcacheD by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1461
* Expose replication lag metrics by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1400
* fix: failing assertion on snapshot regression tests when dbfly instance shuts down by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1463
* fix: support json v1 dot path by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1462
* feat(log): On startup print info to stdout  by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1448
* fix(server): Fix early reply from `DEBUG POPULATE` by @chakaz in https://github.com/dragonflydb/dragonfly/pull/1441
* feat(debian): retain debug symbols in deb package by @Pothulapati in https://github.com/dragonflydb/dragonfly/pull/1464
* fix: avoid using main_listener when migrating connections by @romange in https://github.com/dragonflydb/dragonfly/pull/1471
* fix: Don't calculate replication lags when replica is not in stable sync by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1472
* fix(regression): fix json report filename by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1465
* fix(regression): failing info persistence on snapshot pytests by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1473
* docs: update contributing.md with style guidelines by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1467
* bug(cluster): fix crush on flush slots with shutdown by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1477
* fix: Increase the duration of the seeding in test_replication_info. by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1479
* feat(github regression): add link to github job  by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1482
* cleanups: Remove unused member, rename test, remove unused forward decl. by @chakaz in https://github.com/dragonflydb/dragonfly/pull/1483
* fix(cluster): Fix `CROSSSLOTS` error with multi-key operations by @chakaz in https://github.com/dragonflydb/dragonfly/pull/1488
* feat(cluster): Send flush slots cmd from masters to replicas. by @chakaz in https://github.com/dragonflydb/dragonfly/pull/1484
* feat(server): support command rename by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1489
* Add Japanese README by @eltociear in https://github.com/dragonflydb/dragonfly/pull/1494
* fix: misspelled replication on pytest redis_replication by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1501
* fix(regression): remove test case for info persistence by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1492
* feat: global eval in exec by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1443
* feat: compressed list by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1474
* fix: disable prohibited eval in multi test by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/1505
* fix(server): Initialize ServerFamily with all listeners. by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1485
* fix: avoid crashing dragonfly on parsing error of cgroups config by @romange in https://github.com/dragonflydb/dragonfly/pull/1503
* feat: Support atomic replica takeover by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1314
* chore: tune snyk coverage to ignore test files by @romange in https://github.com/dragonflydb/dragonfly/pull/1509
* Fix a bug and add a timeout test for takeover. by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1512
* fix: null deref in debug log by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1513
* chore: refresh helio by @romange in https://github.com/dragonflydb/dragonfly/pull/1506
* docs : add korean translated README by @brewagebear in https://github.com/dragonflydb/dragonfly/pull/1518
* fix(server): fix prints to log by @adiholden in https://github.com/dragonflydb/dragonfly/pull/1520
* takeover: Cancel blocking commands by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1514
* feat(replication): allow non-tls connections between replica and master #1419  by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1490
* Fix regression test failures on old Python versions by @royjacobson in https://github.com/dragonflydb/dragonfly/pull/1521
* fix: add missing slash on startup for certain log_dir paths by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/1500

## New Contributors
* @wayslog made their first contribution in https://github.com/dragonflydb/dragonfly/pull/1452
* @brewagebear made their first contribution in https://github.com/dragonflydb/dragonfly/pull/1518

## Huge thanks to all the contributors! ❤️

**Full Changelog**: https://github.com/dragonflydb/dragonfly/compare/v1.4.0...v1.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dragonflydb/dragonfly/releases/tag/v1.6.0'>v1.6.0</a>.</em>

**Comment by alexted** [comment]
Guys, I think you have some bug with the release version filling in the announcement post and the release post also. In the title it says  **v1.6.0**, but immediately below it in the next title it says **v1.4.0**. You start to wonder "Am I reading the  release notes of whitch version?".
![Screenshot_20230711_082935_Opera](https://github.com/dragonflydb/dragonfly/assets/33978094/6f7aa3ba-63b6-4386-b28d-e4892bec1ea3)

and

![Screenshot_20230711_083045_Opera](https://github.com/dragonflydb/dragonfly/assets/33978094/4de1a2be-3300-4c68-ba8c-8143f737219c)

**Reply by romange** [reply]
Thanks for letting me know. It was a typo, the new version is 1.6.0 and the release notes are accurate :)
